docs/proposals/koordlet/20230728-support-cold-memory-compute.md

```diff
@@ -0,0 +1,115 @@
+---
+title: Support to collect and report cold memory
+authors:
+  - "@BUPT-wxq"
+reviewers:
+  - "@saintube"
+  - "@zwzhang0107"
+  - "@eahydra"
+  - "@hormes"
+  - "@FillZpp"
+creation-date: 2023-07-28
+last-updated: 2023-07-28
+---
+
+## Table of Contents
+
+[TOC]
+
+## Glossary
+
+**cold page**
+
+A cold page is a page that keeps idle for a long time in the page cache.
+
+**hot page**
+
+A cold page is a page that is often used in the page cache.
+
+**kidled**
+
+Kilded, an open source cold memory collection solution, identifies the hot and cold conditions of nodes, pod and container memory in the cluster. 
+
+## Summary
+
+To support more fine-grained memory overselling, incorporate cold memory compute into memory usage in koordinator.
+
+## Motivation
+
+Currently, the memory usage of nodes and pods collected by koordlet does not contain page cache. It is too aggressive. There are many hot pages that should be counted as memory usage in page cache .That will also help identify cold page and more fine-grained oversell memory. This proposal focuses on cold memory collection and report. 
+
+### Goals
+
+- Define a new util struct named idleinfo  to collect idle memory info.
+- Informer report memory usage of including hot page and crd nodemetric will update status.
+- Collect cold and hot condition  in node, pod and contain and insert metric in metriccache.
+
+### Non-Goals/Future Work
+
+- Cold memory supports  scheduling optimization in koord-manager or koord-scheduler.
+
+## Proposal
+
+### Design
+
+#### collect cold memory info
+
+![image](../../images/support-cold-memory-1.png)
+
+#### report memory usage including hot page
+
+![image](../../images/support-cold-memory-2.png)
+
+
+
+#### add code
+
+- Add a filed in NodeResourceMetric, PodResourceMetric and ContainerResourceMetric named ColdPage to represent cold page size.
+- pkg/koordlet/metriccache/api.go
+
+```go
+type NodeResourceMetric struct {
+	CPUUsed    CPUMetric
```

Comment:
There are indent problems in L76, L93, and L110. Perhaps you are mixing the tab and 4-character space.